Hunted Mobile/Hunted Mobile/Hunted Mobile/ViewModel/MainPageViewModel.cs

```diff
@@ -115,7 +115,13 @@ public class MainPageViewModel : BaseViewModel {
         });
 
         public async Task NavigateToEnterUsernamePage() {
-            await Xamarin.Forms.Application.Current.MainPage.Navigation.PushAsync(new EnterUsername(InviteKeyModel));
+            var viewModel = new EnterUsernameViewModel(inviteKeyModel);
+            var view = new EnterUsername(viewModel);
+            await Xamarin.Forms.Application.Current.MainPage.Navigation.PushAsync(view);
+        }
+
+        public void ResetInviteKey() {
+            InviteKeyModel = new InviteKey();
```

Comment:
Dit hadden we inderdaad wel nodig, maar wordt deze methode wel aangeroepen?

Reply:
Ja in de MainPage code-behind. Daar stond eerst `mainPageViewModel.InviteKeyModel = new InviteKey();` maar omdat dan de view een referentie naar een model (InviteKey) maakt, heb ik deze methode hiervoor gemaakt, dan refereert de view alleen naar die methode van de ViewModel.